manifest: update to remove CONFIG_USB in MCUboot sample

jfischer-no opened the pull request:
MCUboot version update to remove Kconfig option CONFIG_USB.

After the modules are adapted for the revised
USB device stack configuration, and it seems that
it was only necessary for MCUboot, we can finally
remove Kconfig option CONFIG_USB.

jfischer-no committed:
manifest: update to remove CONFIG_USB in MCUboot samples

MCUboot version update to remove Kconfig option CONFIG_USB.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>

jfischer-no committed:
usb: finally remove Kconfig option CONFIG_USB

After the modules are adapted for the revised
USB device stack configuration, and it seems that
it was only necessary for MCUboot, we can finally
remove Kconfig option CONFIG_USB.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>

de-nordic commented:
I am trying to test this with nrf52840dk_nrf52840, with dts modified to allow 64k mcuboot.

I can not upload image to the mcuboot, the mcuboot waits for DFU as expected, but upload process gives following result:
```
root@amdk7:~# dfu-util -l
dfu-util 0.9

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2016 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

Found Runtime: [2fe3:0100] ver=0206, devnum=26, cfg=1, intf=0, path="2-1", alt=0, name="UNKNOWN", serial="7EDDD20AD8B4AF7A"
root@amdk7:~# dfu-util --serial 7EDDD20AD8B4AF7A --download ~debuser/nordic-src/zephyrproject/hello/zephyr/zephyr.bin 
dfu-util 0.9

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2016 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

dfu-util: Invalid DFU suffix signature
dfu-util: A valid DFU suffix will be required in a future dfu-util release!!!
Opening DFU capable USB device...
ID 2fe3:0100
Run-time device DFU version 0110
Claiming USB DFU Runtime Interface...
Determining device status: state = appIDLE, status = 0
Device really in Runtime Mode, send DFU detach request...
Resetting USB...
dfu-util: More than one DFU capable USB device found! Try `--list' and specify the serial number or disconnect all but one device
root@amdk7:~# dfu-util -l
dfu-util 0.9

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2016 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

Found DFU: [2fe3:ffff] ver=0206, devnum=27, cfg=1, intf=0, path="2-1", alt=1, name="UNKNOWN", serial="UNKNOWN"
Found DFU: [2fe3:ffff] ver=0206, devnum=27, cfg=1, intf=0, path="2-1", alt=0, name="UNKNOWN", serial="UNKNOWN"
root@amdk7:~# dfu-util -l
dfu-util 0.9

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2016 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

Found DFU: [2fe3:ffff] ver=0206, devnum=27, cfg=1, intf=0, path="2-1", alt=1, name="UNKNOWN", serial="UNKNOWN"
Found DFU: [2fe3:ffff] ver=0206, devnum=27, cfg=1, intf=0, path="2-1", alt=0, name="UNKNOWN", serial="UNKNOWN"
```

Config and dts patch attached.
[test_config.tar.gz](https://github.com/zephyrproject-rtos/zephyr/files/6931664/test_config.tar.gz)

Tested on Debian win 5.10 AMD64 Kernel.

jfischer-no commented:
>Please advise.

You have to specify Altsetting (image-slot), e.g. --alt 1

```
Found DFU: [2fe3:ffff] ver=0206, devnum=27, cfg=1, intf=0, path="2-1", alt=1, name="UNKNOWN", serial="UNKNOWN"
Found DFU: [2fe3:ffff] ver=0206, devnum=27, cfg=1, intf=0, path="2-1", alt=0, name="UNKNOWN", serial="UNKNOWN"
```

`
dfu-util --serial 7EDDD20AD8B4AF7A --alt 1 --download ~debuser/nordic-src/zephyrproject/hello/zephyr/zephyr.bin `

But event if there is an issue with USB DFU, it is not related to the update as long the device enumerates, what is changed is only how the USB device support is enabled.

de-nordic reviewed (approved):
Tested the change, with the https://github.com/zephyrproject-rtos/zephyr/pull/37453 applied, and seems to compile and work fine (mind the approval comment in https://github.com/zephyrproject-rtos/zephyr/pull/37453).

Looks OK.

jfischer-no committed:
boards: bl654_usb: remove deprecated USB option

Similar to commit 9512ae488a14
("boards: remove USB option for nRF based boards")
Also remove not necessary UART_INTERRUPT_DRIVEN and
duplicate CONFIG_USB_UART_CONSOLE.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>
